lib/matplotlib/axes/_secondary_axes.py

```diff
@@ -0,0 +1,459 @@
+
+        Parameters
+        ----------
+        functions :  2-tuple of func, or `Transform` with an inverse.
+            Transform between the parent axis values and the secondary axis
+            values.
+
+            If supplied as a 2-tuple of functions, the first function is
+            the forward transform function and the second is the inverse
+            transform.
+
+            If a transform is supplied, then the transform must have an
+            inverse.
+
+        """
+
+        if self._orientation == 'x':
+            set_scale = self.set_xscale
+            parent_scale = self._parent.get_xscale()
+        else:
+            set_scale = self.set_yscale
+            parent_scale = self._parent.get_yscale()
+        # we need to use a modified scale so the scale can receive the
+        # transform.  Only types supported are linear and log10 for now.
+        # Probably possible to add other transforms as a todo...
+        if parent_scale == 'log':
+            defscale = 'functionlog'
+        else:
+            defscale = 'function'
+
+        if isinstance(functions, tuple) and callable(functions[0]):
+            # make an arbitrary convert from a two-tuple of functions
+            # forward and inverse.
+            if not callable(functions[1]):
+                raise ValueError('secondary_x/yaxis must have a two-tuple '
+                                 'pair of functions as argument to convert'
+                                 'kwarg.')
+            self._functions = (functions[0], functions[1])
+        elif functions is None:
+            self._functions = (lambda x: x, lambda x: x)
+        else:
+            raise ValueError('functions argument of secondary axes '
+                             'must be a two-tuple of callable functions '
+                             'with the first function being the transform '
+                             'and the second being the inverse.')
+        set_scale(defscale, functions=self._functions)
+
+    def draw(self, renderer=None, inframe=False):
+        """
+        Draw the secondary axes.
+
+        Consults the parent axes for its limits and converts them
+        using the converter specified by
+        `~.axes._secondary_axes.set_functions` (or *functions*
+        parameter when axes initialized.)
+        """
+
+        self._set_lims()
+        # this sets the scale in case the parent has set its scale.
+        self._set_scale()
+        super().draw(renderer=renderer, inframe=inframe)
+
+    def _set_scale(self):
+        """
+        Check if parent has set its scale
+        """
+
+        if self._orientation == 'x':
+            pscale = self._parent.xaxis.get_scale()
+            set_scale = self.set_xscale
+        if self._orientation == 'y':
+            pscale = self._parent.yaxis.get_scale()
+            set_scale = self.set_yscale
+        if pscale == 'log':
+            defscale = 'functionlog'
```

Comment:
Hmm... what about logit, symlog and whatnot?  Can this be made more general?

Reply:
We could add all that.  Not sure if we can make it more general - I’d have to think about that.   When we specify `set_xscale(‘function’, functions=(forward, inverse))` I guess it’d be possible to pass a second scale name?

Reply:
Would be nice to not have to special case log scales (which is what this effectively does), but I don't mind punting this as future work though.

Reply:
I think we could see if anyone misses it for the other scales; I'm guessing this covers 99.9% of use cases...